CompilerSource/gcc_interface/gcc_backend.cpp

```diff
@@ -126,34 +72,26 @@ const char* establish_bearings(const char *compiler)
 
   bool got_success = false;
 
-  // Now we begin interfacing with the toolchain.
-  string cmd, toolchainexec, parameters; // Full command line, executable part, parameter part
-  bool redir; // Whether or not to redirect the output manually
-
   std::string MAKE_paths = compilerInfo.make_vars["PATH"];
   
   std::string dirs = "CODEGEN=" + unixfy_path(codegen_directory) + " ";
   dirs += "WORKDIR=" + unixfy_path(eobjs_directory) + " ";
-  e_execs("make", dirs, "required-directories");
+  actually_bash("make " + dirs + " required-directories");
 
   /* Get a list of all macros defined by our compiler.
   ** These will help us through parsing available libraries.
   ***********************************************************/
-  cmd = compilerInfo.defines_cmd;
-  redir = toolchain_parseout(cmd, toolchainexec,parameters,("\"" + (codegen_directory/"enigma_defines.txt").u8string() + "\""));
-  cout << "Read key `defines` as `" << cmd << "`\nParsed `" << toolchainexec << "` `" << parameters << "`: redirect=" << (redir?"yes":"no") << "\n";
-  got_success = !(redir? e_execsp(toolchainexec, parameters, ("> \"" + (codegen_directory/"enigma_defines.txt").u8string() + "\""),MAKE_paths) : e_execsp(toolchainexec, parameters, MAKE_paths));
-  if (!got_success) return "Call to 'defines' toolchain executable returned non-zero!\n";
+  cout << "Read key `defines` as `" << compilerInfo.defines_cmd << "`" << std::endl;
+  got_success = actually_bash(compilerInfo.defines_cmd + " > " + unixfy_path((codegen_directory).u8string() + "/enigma_defines.txt"));
+  if (got_success != 0) return "Call to 'defines' toolchain executable returned non-zero!\n";
   else cout << "Call succeeded" << endl;
 
   /* Get a list of all available search directories.
   ** These are where we'll look for headers to parse.
   ****************************************************/
-  cmd = compilerInfo.searchdirs_cmd;
-  redir = toolchain_parseout(cmd, toolchainexec,parameters,("\"" + (codegen_directory/"enigma_searchdirs.txt").u8string() + "\""));
-  cout << "Read key `searchdirs` as `" << cmd << "`\nParsed `" << toolchainexec << "` `" << parameters << "`: redirect=" << (redir?"yes":"no") << "\n";
-  got_success = !(redir? e_execsp(toolchainexec, parameters, ("&> \"" + (codegen_directory/"enigma_searchdirs.txt").u8string() + "\""), MAKE_paths) : e_execsp(toolchainexec, parameters, MAKE_paths));
-  if (!got_success) return "Call to 'searchdirs' toolchain executable returned non-zero!";
+  cout << "Read key `searchdirs` as `" << compilerInfo.searchdirs_cmd << "`" << std::endl;
+  got_success = actually_bash(compilerInfo.searchdirs_cmd + " 2> " + unixfy_path((codegen_directory).u8string() + "/enigma_searchdirs.txt"));
```

Comment:
@JoshDreamland idk why tf &>won't work here (it's used in other places) but 2> seems good enough.